tests/src/Kernel/UniqueEmailTest.php

```diff
@@ -0,0 +1,586 @@
+    $user->addRole($role);
+    $violations = $user->validate();
+    $this->assertNotEmpty($violations, 'Decoupled user with role fails validation.');
+
+    // Test validating a coupled user without the role.
+    // Expected result: Violations
+    $user = $this->createUnsavedUser(FALSE, $existing_user->email_prefix);
+    $violations = $user->validate();
+    $this->assertNotEmpty($violations, 'Coupled user without the role fails validation.');
+
+    // Test validating a coupled user with the role.
+    // Expected result: Violations
+    $user = $this->createUnsavedUser(FALSE, $existing_user->email_prefix);
+    $user->addRole($role);
+    $violations = $user->validate();
+    $this->assertNotEmpty($violations, 'Coupled user with the role fails validation.');
+  }
+
+  /**
+   * Test the unique email validator in 'include' mode with an existing coupled
+   * user with no matching roles.
+   *
+   * @covers ::validate
+   */
+  public function testModeIncludeCoupledWithoutRole() {
+    $role = $this->createRole([]);
+    $this->setUniqueEmailsConfig('include', [$role]);
+
+    $existing_user = $this->createUser();
+
+    // Test validating a coupled user without the role.
+    // Expected result: No violations
+    $user = $this->createUnsavedUser(TRUE, $existing_user->getAccountName());
+    $violations = $user->validate();
+    $this->assertEmpty($violations, 'Decoupled user without role passes validation.');
+
+    // Test validating a coupled user with the role.
+    // Expected result: Violations
+    $user = $this->createUnsavedUser(TRUE, $existing_user->getAccountName());
+    $user->addRole($role);
+    $violations = $user->validate();
+    $this->assertNotEmpty($violations, 'Decoupled user with role fails validation.');
+
+    // Test validating a coupled user without the role.
+    // Expected result: Violations
+    $user = $this->createUnsavedUser(FALSE, $existing_user->getAccountName());
+    $violations = $user->validate();
+    $this->assertNotEmpty($violations, 'Coupled user without the role fails validation.');
+
+    // Test validating a coupled user with the role.
+    // Expected result: Violations
+    $user = $this->createUnsavedUser(FALSE, $existing_user->getAccountName());
+    $user->addRole($role);
+    $violations = $user->validate();
+    $this->assertNotEmpty($violations, 'Coupled user with the role fails validation.');
+  }
+
+  /**
+   * Test the unique email validator in 'include' mode with an existing coupled
+   * user with a matching roles.
+   *
+   * @covers ::validate
+   */
+  public function testModeIncludeCoupledWithRole() {
+    $role = $this->createRole([]);
+    $this->setUniqueEmailsConfig('include', [$role]);
+
+    $existing_user = $this->createUser();
+    $existing_user->addRole($role);
+    $existing_user->save();
+
+    // Test validating a coupled user without the role.
+    // Expected result: No violations
+    $user = $this->createUnsavedUser(TRUE, $existing_user->getAccountName());
```

Comment:
Comment and message is wrong? OR the createUnsavedUser call is wrong? It says its testing a coupled user but is actually testing a decoupled.